src/language_server/vfs.rs

```diff
@@ -0,0 +1,153 @@
+use std::collections::HashMap;
+use std::io;
+use std::io::ErrorKind;
+use std::sync::RwLock;
+
+use super::document::Document;
+
+use reqwest::blocking::Client;
+use tower_lsp::lsp_types::Url;
+
+#[derive(Debug)]
+pub struct VFS {
+    overlays: RwLock<HashMap<Url, RwLock<Document>>>,
+
+    http_client: Client,
+}
+
+impl VFS {
+    pub fn new() -> io::Result<VFS> {
+        let client = Client::builder()
+            .gzip(true)
+            .build()
+            .map_err(categorise_reqwest_error)?;
+
+        Ok(VFS {
+            overlays: RwLock::new(HashMap::new()),
+            http_client: client,
+        })
+    }
+
+    pub fn get_document_contents(&self, uri: &Url) -> io::Result<String> {
+        {
+            let hm = self.overlays.read().unwrap();
```

Comment:
Let's remove the `unwrap`s in this file. 

I think they will trigger if a mutex is poisoned, which is a compiler bug rather than a user error so we can use `.gleam_expect`.